ufo2fontir/src/source.rs

```diff
@@ -700,25 +700,18 @@ impl Work<Context, WorkId, WorkError> for StaticMetadataWork {
 
         let tags_by_name = axes.iter().map(|a| (a.name.as_str(), a.tag)).collect();
         let axes_by_tag = axes.iter().map(|a| (a.tag, a)).collect();
-        let family_prefix = names
-            .get(&NameKey::new_bmp_only(NameId::FAMILY_NAME))
-            .map(|name| name.clone() + " ")
-            .unwrap_or_default();
         let named_instances = self
             .designspace
             .instances
             .iter()
-            .map(|inst| NamedInstance {
-                name: match inst
-                    .name
-                    .as_ref()
-                    .unwrap()
-                    .strip_prefix(family_prefix.as_str())
-                {
-                    Some(tail) => tail.to_string(),
-                    None => inst.name.clone().unwrap(),
-                },
-                location: to_design_location(&tags_by_name, &inst.location).to_user(&axes_by_tag),
+            .map(|inst| {
+                // TODO: Also support localised names, and names inferred from axis labels
+                // (also used to build STAT table)
+                NamedInstance {
+                    name: inst.stylename.clone().unwrap_or(inst.name.clone().unwrap()),
```

Comment:
initially here I had a try_fold that would return an error when stylename was not set, but then I thought it wasn't worth the hassle and I'm falling back to the instance.name that we have already populated with `unnamed_instance_{0}`, etc. upon loading the designspace and thus we can unwrap().

Reply:
I actually ended up restoring the family-prefix stipping of instance.name which I had initially removed, even though it doesn't match fonttools, may still have uses. In any case we shall revisit this when we will add support for instance stylenames from DSv5 axis labels